macros/src/codegen.rs

```diff
@@ -279,7 +260,7 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
                     guard_set.push(guard.clone());
                     guard_list.extend(quote! {
                         #[allow(missing_docs)]
-                        fn #guard_with_lifetimes(&mut self, #temporary_context #state_data #event_data) -> Result<(), #guard_error>;
+                        fn #guard <#all_lifetimes_code> (&mut self, #temporary_context #state_data #event_data) -> Result<(), #guard_error>;
```

Comment:
Is this and below going to generate warnings about unused lifetime specifiers?

Reply:
No. This is how it is also used on `Events` and `States` definition.